File: 2nd Year/1st Semester/MAP/Seminar/Seminar10/Seminar10/Domain/Pontaj.cs
using System.Runtime.InteropServices.ComTypes;

namespace Seminar10.Domain;

public class Pontaj
{
    private Angajat angajat { set; get; }
    private Sarcina sarcina { set; get; }
    private DateTime data;

    public Pontaj(DateTime data, Angajat angajat, Sarcina sarcina)
    {
        this.data = data;
        this.angajat = angajat;
        this.sarcina = sarcina;
    }

    public override string ToString()
    {
        return $"{nameof(data)}: {data}, {nameof(angajat)}: {angajat}, {nameof(sarcina)}: {sarcina}";
    }
}

File: 2nd Year/1st Semester/MAP/Seminar/Sem13-MAP/src/main/java/ubb/scs/map/sem13map/domain/Pontaj.java
package ubb.scs.map.sem13map.domain;

import java.time.LocalDateTime;

public class Pontaj extends  Entity<Tuple<String>> {
    private LocalDateTime data;
    private Angajat angajat;
    private Sarcina sarcina;

    public Pontaj(LocalDateTime data, Angajat angajat, Sarcina sarcina) {
        this.data = data;
        this.angajat = angajat;
        this.sarcina = sarcina;
        setId(new Tuple<>(angajat.getId(), sarcina.getId()));
    }

    public LocalDateTime getData() {
        return data;
    }

    public void setData(LocalDateTime data) {
        this.data = data;
    }

    public Angajat getAngajat() {
        return angajat;
    }

    public void setAngajat(Angajat angajat) {
        this.angajat = angajat;
    }

    public Sarcina getSarcina() {
        return sarcina;
    }

    public void setSarcina(Sarcina sarcina) {
        this.sarcina = sarcina;
    }

    @Override
    public String toString() {
        return "Pontaj{" +
                "data=" + data +
                ", angajat=" + angajat +
                ", sarcina=" + sarcina +
                '}';
    }
}
